include/deal.II/lac/tensor_product_matrix.templates.h

```diff
@@ -0,0 +1,86 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2022 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+#ifndef dealii_tensor_product_matrix_templates_h
+#define dealii_tensor_product_matrix_templates_h
+
+
+#include <deal.II/base/config.h>
+
+#include <deal.II/lac/tensor_product_matrix.h>
+
+#ifndef FDM_DEGREE_MAX
+// We set this value 17, since this is the value needed for smoothers for
+// cell-centered patches with overlap for constinuous elements with degrees up
+// to FE_EVAL_FACTORY_DEGREE_MAX.
+#  define FDM_DEGREE_MAX 17
```

Comment:
This should be coupled to 
https://github.com/dealii/dealii/blob/dfde63a25cf241f8ef3e67f8e247258cd84de634/include/deal.II/matrix_free/evaluation_template_factory_internal.h#L24

Related to https://github.com/dealii/dealii/issues/14106.